cmake/JSSTests.cmake

```diff
@@ -0,0 +1,200 @@
+        "org.mozilla.jss.tests.HmacTest;${RESULTS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "KeyWrapping"
+        "org.mozilla.jss.tests.JCAKeyWrap;${RESULTS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "Mozilla_JSS_JCA_Signature"
+        "org.mozilla.jss.tests.JCASigTest;${RESULTS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "Mozilla_JSS_NSS_Signature"
+        "org.mozilla.jss.tests.SigTest;${RESULTS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "JSS_Signature_test"
+        "org.mozilla.jss.tests.SigTest;${RESULTS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "Secret_Decoder_Ring"
+        "org.mozilla.jss.tests.TestSDR;${RESULTS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "List_cert_by_certnick"
+        "org.mozilla.jss.tests.ListCerts;${RESULTS_OUTPUT_DIR};Server_RSA"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "Verify_cert_by_certnick"
+        "org.mozilla.jss.tests.VerifyCert;${RESULTS_OUTPUT_DIR};${PASSWORD_FILE};Server_RSA"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "Secret_Key_Generation"
+        "org.mozilla.jss.tests.SymKeyGen;${RESULTS_OUTPUT_DIR}"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "Mozilla_JSS_Secret_Key_Generation"
+        "org.mozilla.jss.tests.JCASymKeyGen;${RESULTS_OUTPUT_DIR}"
+        "Setup_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "Enable_FipsMODE"
+        "org.mozilla.jss.tests.FipsTest;${RESULTS_FIPS_OUTPUT_DIR};enable"
+        "Setup_FIPS_DBs" # DEPENDS
+    )
+    jss_test_java(
+        "check_FipsMODE"
+        "org.mozilla.jss.tests.FipsTest;${RESULTS_FIPS_OUTPUT_DIR};chkfips"
+        "Enable_FipsMODE" # DEPENDS
+    )
+    jss_test_java(
+        "HMAC_FIPSMODE"
+        "org.mozilla.jss.tests.HMACTest;${RESULTS_FIPS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "check_FipsMODE" # DEPENDS
+    )
+    jss_test_java(
+        "KeyWrapping_FIPSMODE"
+        "org.mozilla.jss.tests.JCAKeyWrap;${RESULTS_FIPS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "HMAC_FIPSMODE" # DEPENDS
+    )
+    jss_test_java(
+        "Mozilla_JSS_JCA_Signature_FIPSMODE"
+        "org.mozilla.jss.tests.JCASigTest;${RESULTS_FIPS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "KeyWrapping_FIPSMODE" # DEPENDS
+    )
+    jss_test_java(
+        "JSS_Signature_test_FipsMODE"
+        "org.mozilla.jss.tests.SigTest;${RESULTS_FIPS_OUTPUT_DIR};${PASSWORD_FILE}"
+        "Mozilla_JSS_JCA_Signature_FIPSMODE" # DEPENDS
+    )
+    jss_test_java(
+        "Disable_FipsMODE"
+        "org.mozilla.jss.tests.FipsTest;${RESULTS_FIPS_OUTPUT_DIR};disable"
+        "JSS_Signature_test_FipsMODE" # DEPENDS
+    )
+endmacro()
+
+function(jss_test_java TEST_NAME TEST_COMMAND)
```

Comment:
This is fine but for future improvement it would be nice if we can use keywords to separate the arguments like this:
```
jss_test_java(
    NAME
        Disable_FipsMODE"
    COMMAND
        "org.mozilla.jss.tests.FipsTest;${RESULTS_FIPS_OUTPUT_DIR};disable"
    DEPENDS
        JSS_Signature_test_FipsMODE
)
```

Reply:
Here's an example:
https://github.com/dogtagpki/pki/blob/master/cmake/Modules/Java.cmake#L21
But probably it can be done more easily with CMakeParseArguments().

Reply:
Looks good!

Reply:
Thanks for the suggestion, didn't know that existed and makes it more readable. :)